packages/dds/matrix/src/runtime.ts

```diff
@@ -49,7 +49,7 @@ export class SharedMatrixFactory implements IChannelFactory<ISharedMatrix> {
 		return matrix;
 	}
 
-	public create(document: IFluidDataStoreRuntime, id: string): ISharedMatrix & IChannel {
+	public create(document: IFluidDataStoreRuntime, id: string): SharedMatrix & IChannel {
```

Comment:
why this change? this seems like a step in the wrong direction--we'd like to apply the pattern we've done for SharedDirectory/SharedMap (hide the implementation class, expose only the interface) to all other DDSes.

I feel similarly to the changes in sequence-deprecated, but it's less of a big deal there since it's not clear anyone uses that code in production and our support guarantee for it is basically nonexistent.

Reply:
This is called out in the PR description: "Make concreate internal types return more specific types in a few places (makes having them expose internal types and having their tests use those types easier)."

Generally I find having internal concreate code return more specific types makes testing within the package easier.

Here the factory class is package exported, but only as alpha since it should be replaced by ISharedObjectKind<IMatrix> which will make this factory have no impact on the package api. The tests in this package however would like to work with the actual concrete type and this allows them to do so without a cast. THis is handle as it allows the concreate class for the DDS to return more specific types (in this case internal handle interfaces once handle is split) allowing for easier testing.

If you would prefer this change not be done until the alpha export of this factory class is removed, thats fine and the tests can instead down cast the factory output as needed.

Reply:
Reverted.